### PR TITLE
Adding video share to Instagram Stories 

### PR DESCRIPTION
**Pull request by slaycky**
add method video to instagram stories.

### CODE REVIEW

**Comment by rafaelcamaram** [comment]
It’s an really good improvement. I was looking for something like this! 

I hope this PR is approved

File: ios/InstagramStories.m [hunk 1]
```diff
@@ -82,17 +82,36 @@ - (void)backgroundImage:(NSData *)backgroundImage stickerImage:(NSData *)sticker
...
     successCallback:(RCTResponseSenderBlock)successCallback {
     
     NSString *attrURL = [RCTConvert NSString:options[@"attributionURL"]];
+    //NSData *backgroundVideo = [[NSFileManager defaultManager] contentsAtPath:path];
     if (attrURL == nil) {
         attrURL = @"";
     }
     
     NSString *method = [RCTConvert NSString:options[@"method"]];
+       RCTLog(@"method method is");
```
Comment (jgcmarins) [inline ios/InstagramStories.m]: can we remove this?

**Comment by jgcmarins** [comment]
@rafaelcamaram good to know!
I am also excited with this new feature.
Could you please help us testing this PR?

File: ios/InstagramStories.m (hunk 1 above)
Comment (slaycky) [reply ios/InstagramStories.m]: Of Course! I 'll remove and make a new request! Thanks.

**Comment by MateusAndrade** [comment]
> We also implemented this feature. It would be great to have it in the module

Can you confirm is this PR is working? 🤔 I forced an update with our master, but it would be nice to test this on real life case.

**Comment by alexandruenache19** [comment]
> > We also implemented this feature. It would be great to have it in the module
> 
> Can you confirm is this PR is working? 🤔 I forced an update with our master, but it would be nice to test this on real life case.

We have not tested this PR specifically but our solution is the same and it works fine on Iphone 11. OS Version 13.x. We were able to share a video as Insta Story

File: ios/InstagramStories.m [hunk 2]
```diff
@@ -82,12 +82,30 @@ - (void)backgroundImage:(NSData *)backgroundImage stickerImage:(NSData *)sticker
...
+    }
+}
+
 
 - (void)shareSingle:(NSDictionary *)options
     failureCallback:(RCTResponseErrorBlock)failureCallback
     successCallback:(RCTResponseSenderBlock)successCallback {
     
     NSString *attrURL = [RCTConvert NSString:options[@"attributionURL"]];
+    //NSData *backgroundVideo = [[NSFileManager defaultManager] contentsAtPath:path];
```
Comment (rpavlovs) [inline ios/InstagramStories.m:108]: 💣

File: ios/InstagramStories.m [hunk 3]
```diff
@@ -135,9 +153,17 @@ - (void)shareSingle:(NSDictionary *)options
             } else {
                 UIImage *backgroundImage = [UIImage imageWithData: [NSData dataWithContentsOfURL:backgroundURL]];
                 UIImage *stickerImage = [UIImage imageWithData: [NSData dataWithContentsOfURL:sticketURL]];
-                
                 [self backgroundImage:UIImagePNGRepresentation(backgroundImage) stickerImage:UIImagePNGRepresentation(stickerImage) attributionURL:attrURL];
             }
+        }else if([method isEqualToString:@"shareBackgroundVideo"]) {
```
Comment (rpavlovs) [inline ios/InstagramStories.m:158]:
```suggestion
        } else if([method isEqualToString:@"shareBackgroundVideo"]) {
```

File: ios/InstagramStories.m [hunk 4]
```diff
@@ -82,12 +82,30 @@ - (void)backgroundImage:(NSData *)backgroundImage stickerImage:(NSData *)sticker
...
 
+- (void)backgroundVideo:(NSData *)backgroundVideo
+{
+    // Verify app can open custom URL scheme. If able,
+    // assign assets to pasteboard, open scheme.
+    NSURL *urlScheme = [NSURL URLWithString:@"instagram-stories://share"];
+    if ([[UIApplication sharedApplication] canOpenURL:urlScheme]) {
+        // Assign background image asset and attribution link URL to pasteboard
+        NSArray *pasteboardItems = @[@{@"com.instagram.sharedSticker.backgroundVideo" : backgroundVideo}];
+        //NSArray *pasteboardItems = @[@{@"com.instagram.sharedSticker.backgroundVideo" : backgroundVideo;
```
Comment (rpavlovs) [inline ios/InstagramStories.m:93]:
```suggestion
```

**Comment by rpavlovs** [comment]
It works fine on iPhone 10, 13.4.1 👍

**Comment by rpavlovs** [comment]
One thing missing is support for `backgroundTopColor`/`backgroundBottomColor`, but it's missing for Image as well, so probably best to add support for both in a separate PR.

**Comment by MateusAndrade** [comment]
@rpavlovs I think we already have something similar in the current version of this library. Would you mind checking: https://github.com/react-native-community/react-native-share#static-values-for-instagram-stories. 

Maybe we can use some improvements in this PR as well. 🤔

**Comment by rpavlovs** [comment]
Looks like background colours are set for stickers only: https://github.com/react-native-community/react-native-share/blob/master/ios/InstagramStories.m#L107-L127

Not sure if we end up using share to insta stories, because of the 20 sec video limit (we need 1 min), but I'll make those changes if we do.

**Comment by jgcmarins** [comment]
@slaycky would you mind to fix conflicts? Thanks.

**Comment by alexandruenache19** [comment]
Any update? We really need this feature. Any way I can help to do it faster? I am new to how this process works

**Comment by mikehardy** [comment]
@alexandruenache19 right now it looks like there is a big conflict in the README which @slaycky needs to resolve then it might be good to go - @slaycky hasn't responded in a while though, if you made a fresh PR just in case, with everything cleaned and resolved based off this one and confirmed it worked, it might be merge-able?

**Comment by MateusAndrade** [comment]
> @alexandruenache19 right now it looks like there is a big conflict in the README which @slaycky needs to resolve then it might be good to go - @slaycky hasn't responded in a while though, if you made a fresh PR just in case, with everything cleaned and resolved based off this one and confirmed it worked, it might be merge-able?

We can follow this approach if you feel comfortable with that @alexandruenache19. I only think that we need to give the credits to @slaycky for that.

**Comment by slaycky** [comment]
Hello Guys, Sorry for not responding at time. I resolved the conflict. waiting for 1 approving

**Comment by MateusAndrade** [comment]
> Hello Guys, Sorry for not responding at time. I resolved the conflict. waiting for 1 approving

Actually we to move these changes on README.MD explaining this to our [docs](https://github.com/react-native-community/react-native-share/tree/master/website). Let me know if you need any help. 😄

File: README.md [hunk 5]
```diff
@@ -39,8 +39,396 @@ Then simply import:
...
+| Name  | Type     | Description |
+| :---- | :------: | :--- |
+| appId | string   | (required) facebook appId |
+| backgroundImage | string   | URL you want to share |
+| stickerImage | string   | URL you want to share |
+| method | string   | [List](#instagram-stories-method-list) |
+| backgroundBottomColor | string   |  (optional) default #837DF4 |
+| backgroundTopColor | string   | (optional) default #906df4 |
+| attributionURL | string   | (optional) facebook beta-test |
+
```
Comment (MateusAndrade) [inline README.md:428]: We need to place this in the desired location at our docs

File: ios/InstagramStories.m (hunk 4 above)
Comment (MateusAndrade) [inline ios/InstagramStories.m:93]: Remove this comment

File: ios/InstagramStories.m (hunk 2 above)
Comment (MateusAndrade) [inline ios/InstagramStories.m:108]: Remove commented line

File: ios/InstagramStories.m (hunk 3 above)
Comment (MateusAndrade) [reply ios/InstagramStories.m:158]: We can apply that @slaycky ?

File: ios/InstagramStories.m (hunk 2 above)
Comment (slaycky) [reply ios/InstagramStories.m]: removed

File: ios/InstagramStories.m (hunk 4 above)
Comment (slaycky) [reply ios/InstagramStories.m]: removed

File: README.md (hunk 5 above)
Comment (MateusAndrade) [reply README.md:428]: @slaycky do you need help with that? If you have any doubt we are using docusarus to our docs. You can read more about that [here](https://react-native-community.github.io/react-native-share/docs/contributing)

Comment (slaycky) [reply README.md]: Thank you @MateusAndrade, I removed the lines, Can you check please if it's right?

Comment (MateusAndrade) [reply README.md]: The README.MD is right now, but we still need to add these changes to the DOCS at: [share-open](https://react-native-community.github.io/react-native-share/docs/share-open) or [share-single](https://react-native-community.github.io/react-native-share/docs/share-single). Do you need help with that?

File: ios/InstagramStories.m [hunk 6]
```diff
@@ -135,9 +151,17 @@ - (void)shareSingle:(NSDictionary *)options
             } else {
                 UIImage *backgroundImage = [UIImage imageWithData: [NSData dataWithContentsOfURL:backgroundURL]];
                 UIImage *stickerImage = [UIImage imageWithData: [NSData dataWithContentsOfURL:sticketURL]];
-                
                 [self backgroundImage:UIImagePNGRepresentation(backgroundImage) stickerImage:UIImagePNGRepresentation(stickerImage) attributionURL:attrURL];
             }
+        } else if([method isEqualToString:@"shareBackgroundVideo"]) {
+            
+            NSString *URL = [RCTConvert NSString:options[@"backgroundVideo"]];
```
Comment (MateusAndrade) [inline ios/InstagramStories.m:158]: One question: we need to pass a URL to the video or this will work with a base64?

Comment (slaycky) [reply ios/InstagramStories.m:158]: Need to pass URL. I'm not to sure about base64

File: README.md (hunk 5 above)
Comment (slaycky) [reply README.md]: Hello @MateusAndrade can ou help me? I have to change this file https://github.com/react-native-community/react-native-share/compare/master...slaycky:patch-2?

Comment (MateusAndrade) [reply README.md]: Yes! This one: https://react-native-community.github.io/react-native-share/docs/share-single or https://github.com/react-native-community/react-native-share/edit/master/website/docs/share-single.mdx.

**Comment by vforvasile** [comment]
@MateusAndrade, I noticed the commits reflect only iOS but the docs don't mention any limitation on Android, will this param `backgroundVideo ` work on Android as well?

**Comment by MateusAndrade** [comment]
> @MateusAndrade, I noticed the commits reflect only iOS but the docs don't mention any limitation on Android, will this param `backgroundVideo ` work on Android as well?

I did a quick search and it looks that it's working on iOS only. I think we need to add a new column regarding of the limitation of the options passed when sharing with Instagram Stories. Which can be seen [here](https://react-native-community.github.io/react-native-share/docs/share-single#supported-options-for-instagram_stories).

Would you mind helping us with that @vforvasile ? 😅

**Pull request merged**